ext/mixed/js/display-generator.js

```diff
@@ -498,10 +499,22 @@ class DisplayGenerator {
                         dictionaryResults.push(pitchInfo);
                     }
                     pitchInfo.expressions.add(expression);
+                    allExpressions.add(expression);
                 }
             }
         }
 
+        for (const dictionaryResults of results.values()) {
+            for (const result of dictionaryResults) {
+                const exclusiveExpressions = [];
+                const resultExpressions = result.expressions;
+                if (!areSetsEqual(resultExpressions, allExpressions)) {
+                    exclusiveExpressions.push(...getSetIntersection(resultExpressions, allExpressions));
+                }
+                result.exclusiveExpressions = exclusiveExpressions;
```

Comment:
I'd update it like this to handle readings as well:
```diff
diff --git a/ext/mixed/js/display-generator.js b/ext/mixed/js/display-generator.js
index 8177c31c..b6469d62 100644
--- a/ext/mixed/js/display-generator.js
+++ b/ext/mixed/js/display-generator.js
@@ -482,10 +482,13 @@ class DisplayGenerator {
         const results = new Map();

         const allExpressions = new Set();
+        const allReadings = new Set();
         const expressions = definition.expressions;
         const sources = Array.isArray(expressions) ? expressions : [definition];
         for (const {pitches: expressionPitches, expression} of sources) {
+            allExpressions.add(expression);
             for (const {reading, pitches, dictionary} of expressionPitches) {
+                allReadings.add(reading);
                 let dictionaryResults = results.get(dictionary);
                 if (typeof dictionaryResults === 'undefined') {
                     dictionaryResults = [];
@@ -499,7 +502,6 @@ class DisplayGenerator {
                         dictionaryResults.push(pitchInfo);
                     }
                     pitchInfo.expressions.add(expression);
-                    allExpressions.add(expression);
                 }
             }
         }
@@ -508,9 +510,13 @@ class DisplayGenerator {
             for (const result of dictionaryResults) {
                 const exclusiveExpressions = [];
                 const resultExpressions = result.expressions;
+                const resultReadings = new Set([result.reading]);
                 if (!areSetsEqual(resultExpressions, allExpressions)) {
                     exclusiveExpressions.push(...getSetIntersection(resultExpressions, allExpressions));
                 }
+                if (!areSetsEqual(resultReadings, allReadings)) {
+                    exclusiveExpressions.push(...getSetIntersection(resultReadings, allReadings));
+                }
                 result.exclusiveExpressions = exclusiveExpressions;
             }
         }
```

Reply:
> I'd update it like this to handle readings as well:

The readings are redundant when the downstep notation is also visible. Does it make sense to not display them for this case? (I think this is why I initially didn't add this, because I forgot that there were options to turn that off.)

![image](https://user-images.githubusercontent.com/11037431/77851884-8c91d280-71a9-11ea-8f6d-ac892dcd118c.png)

Reply:
Yeah, I think it's good to add that exception.

Reply:
Updated.